### DISCUSSION TITLE
Need help with next js redirect

### DISCUSSION THREAD

**Discussion by PaisaDekho** [vercel/next.js#52770] [App Router]
Hello everyone , so for my company I'm using a payment gateway for that i first have to send them a encrypted request which then will redirect my users to their payment gateway page and after a successful payment they'll redirect my user to the link which i send them through the request but for some reason when they come back to my page it shows this page doesn't exist but  i can still access that route normally .

**Comment by icyJoseph** [comment]
- Does reloading the page on this route work?
- Are you seeing an actual 404?
- Are they changing the return url route in any way?

**Comment by PaisaDekho** [comment]
I won't say it works but there's a request coming with the page I can't capture it , it shows this page but if i make a request again then it shows a normal page in my case it's just a simple text this is the payment handler but by then the encrypted  request is already destroyed . 


<img width="1552" alt="Screenshot 2023-07-17 at 3 17 07 PM" src="https://github.com/vercel/next.js/assets/125072569/640b597c-be7b-46a2-a3f2-c5509ca914d2">

**Reply by icyJoseph** [reply]
Could you describe the flow step by step? Like,

1. user goes to payment handler
2. a request is sent to the 3rd party
3. 3rd party respond with a redirect location header 301, 302, 307
4. user completes payment
5. 3rd party redirects back to payment handler

I wonder if the page is getting marked a non-interactive because it just redirects.

**Reply by rohitshoogloo** [reply]
Hi, I am also stuck with this same issue. Did you found any solution for this error ?

**Comment by PaisaDekho** [comment]
1.user clicks on pay
2. we sent them to payment gateway site with all the necessary information amount , success url , cancel url etc.
3. they do their payment and then the payment gateway redirect the user to that success_url in my case it's /paymentHandler
4. now I have to capture the request body that comes with that request 
5. but it shows the error you can see above 

it might be because unresponsive  because it's redirecting . because in Next js 13 i also tried redirecting with middleware but it gave me the same error 

the is the code which was giving me a error 

```
export default function middleware(request: NextRequest) {
  if (!request.cookies.get("token")?.value) {
    return NextResponse.rewrite(new URL("/login", request.url));
  }

  return NextResponse.next();
}
```
anyway i did it using next js config redirect later

**Reply by rohitshoogloo** [reply]
Hi, I am also stuck with this same issue. Did you found any solution for this error ? 

In my case body always returning undefined.
